cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -68,22 +72,92 @@ public URI getRootUri() {
         return this.uriHelper.getRootUri();
     }
 
+    // result of executing an HTTP call:
+    // - stream non-null and exception null: the call was successful, result in stream
+    // - stream null and exception non-null: the call was unsuccessful, details in exception
+    // - fatal: set to true when exception non-null, indicates call should not be retried
+    private class ExecuteResult
```

Comment:
Perhaps there is an argument for this to be extracted to a top level class and potentially used as the return type for the `HttpConnection#execute()`. Although I consider that refactor out of scope for this PR.